diff --git a/src/wrap/wrap_close.c b/src/wrap/wrap_close.c
index 83e4e92..5a3c888 100644
--- a/src/wrap/wrap_close.c
+++ b/src/wrap/wrap_close.c
@@ -6,7 +6,7 @@
 /*   By: Juyeong Maing <jmaing@student.42seoul.kr>  +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/09/04 19:09:42 by Juyeong Maing     #+#    #+#             */
-/*   Updated: 2022/09/04 19:10:52 by Juyeong Maing    ###   ########.fr       */
+/*   Updated: 2022/09/07 06:06:39 by Juyeong Maing    ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
@@ -14,7 +14,12 @@
 
 #include <fcntl.h>
 
+#include <stdio.h>
+
 int	wrap_close(int fd)
 {
-	return (close(fd));
+	const int	result = close(fd);
+
+	fprintf(stderr, "close(%d) = %d\n", fd, result);
+	return (result);
 }
diff --git a/src/wrap/wrap_open.c b/src/wrap/wrap_open.c
index b812b03..10c9974 100644
--- a/src/wrap/wrap_open.c
+++ b/src/wrap/wrap_open.c
@@ -6,7 +6,7 @@
 /*   By: Juyeong Maing <jmaing@student.42seoul.kr>  +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/09/04 19:09:42 by Juyeong Maing     #+#    #+#             */
-/*   Updated: 2022/09/04 19:10:34 by Juyeong Maing    ###   ########.fr       */
+/*   Updated: 2022/09/07 06:21:12 by Juyeong Maing    ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
@@ -14,7 +14,12 @@
 
 #include <fcntl.h>
 
+#include <stdio.h>
+
 int	wrap_open(const char *path, int flags)
 {
-	return (open(path, flags));
+	const int	result = open(path, flags);
+
+	fprintf(stderr, "open(\"%s\", %#x) = %d\n", path, flags, result);
+	return (result);
 }
diff --git a/src/wrap/wrap_open2.c b/src/wrap/wrap_open2.c
index 08e021a..123c0a4 100644
--- a/src/wrap/wrap_open2.c
+++ b/src/wrap/wrap_open2.c
@@ -6,15 +6,21 @@
 /*   By: Juyeong Maing <jmaing@student.42seoul.kr>  +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/09/04 19:09:42 by Juyeong Maing     #+#    #+#             */
-/*   Updated: 2022/09/04 19:53:53 by Juyeong Maing    ###   ########.fr       */
+/*   Updated: 2022/09/07 06:21:06 by Juyeong Maing    ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "wrap.h"
 
+#include <stdio.h>
+
 #include <fcntl.h>
 
 int	wrap_open2(const char *path, int flags, int additional_flags)
 {
-	return (open(path, flags, additional_flags));
+	const int	result = open(path, flags, additional_flags);
+
+	fprintf(stderr, "open(\"%s\", %#x, %#o) = %d\n",
+		path, flags, additional_flags, result);
+	return (result);
 }
diff --git a/src/wrap/wrap_pipe.c b/src/wrap/wrap_pipe.c
index 05085f1..6e37afd 100644
--- a/src/wrap/wrap_pipe.c
+++ b/src/wrap/wrap_pipe.c
@@ -6,15 +6,19 @@
 /*   By: Juyeong Maing <jmaing@student.42seoul.kr>  +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/09/04 19:11:09 by Juyeong Maing     #+#    #+#             */
-/*   Updated: 2022/09/04 19:11:31 by Juyeong Maing    ###   ########.fr       */
+/*   Updated: 2022/09/07 06:07:50 by Juyeong Maing    ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "wrap.h"
 
 #include <unistd.h>
+#include <stdio.h>
 
 int	wrap_pipe(int fds[2])
 {
-	return (pipe(fds));
+	const int	result = pipe(fds);
+
+	fprintf(stderr, "pipe(_) = %d (%d, %d)\n", result, fds[0], fds[1]);
+	return (result);
 }
diff --git a/src/wrap/wrap_read.c b/src/wrap/wrap_read.c
index 49e6d91..5006898 100644
--- a/src/wrap/wrap_read.c
+++ b/src/wrap/wrap_read.c
@@ -6,15 +6,19 @@
 /*   By: Juyeong Maing <jmaing@student.42seoul.kr>  +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/09/04 19:09:00 by Juyeong Maing     #+#    #+#             */
-/*   Updated: 2022/09/04 20:03:49 by Juyeong Maing    ###   ########.fr       */
+/*   Updated: 2022/09/07 06:04:46 by Juyeong Maing    ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "wrap.h"
 
 #include <unistd.h>
+#include <stdio.h>
 
 ssize_t	wrap_read(int fd, void *buffer, size_t bytes)
 {
-	return (read(fd, buffer, bytes));
+	const ssize_t	result = read(fd, buffer, bytes);
+
+	fprintf(stderr, "read(%d, _, %zu) = %zd\n", fd, bytes, result);
+	return (result);
 }
diff --git a/src/wrap/wrap_write.c b/src/wrap/wrap_write.c
index 4d83e28..ebe3245 100644
--- a/src/wrap/wrap_write.c
+++ b/src/wrap/wrap_write.c
@@ -6,7 +6,7 @@
 /*   By: Juyeong Maing <jmaing@student.42seoul.kr>  +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/09/04 19:09:00 by Juyeong Maing     #+#    #+#             */
-/*   Updated: 2022/09/04 20:03:52 by Juyeong Maing    ###   ########.fr       */
+/*   Updated: 2022/09/07 06:18:23 by Juyeong Maing    ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
@@ -14,7 +14,12 @@
 
 #include <unistd.h>
 
+#include <stdio.h>
+
 ssize_t	wrap_write(int fd, const void *buffer, size_t bytes)
 {
-	return (write(fd, buffer, bytes));
+	const ssize_t	result = write(fd, buffer, bytes);
+
+	fprintf(stderr, "write(%d, _, %zu) = %zd\n", fd, bytes, result);
+	return (result);
 }
